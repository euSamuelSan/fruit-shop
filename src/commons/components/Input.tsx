import React from 'react'
import styled from 'styled-components'
import Colors from '../Colors'

export type InputProps = {
    placeholder?: string
    className?: string
    value: string
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
    type?: string
}

export default function Input({
    placeholder,
    className,
    value,
    onChange,
    type = 'text',
}: InputProps): JSX.Element {
    return (
        <BasicInput
            type={type}
            placeholder={placeholder}
            className={className}
            value={value}
            onChange={onChange}
        />
    )
}

const BasicInput = styled.input`
    outline: none;
    padding: 10px;
    border: none;

    color: ${Colors.black};
`
