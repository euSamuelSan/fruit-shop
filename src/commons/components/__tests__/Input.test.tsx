import { fireEvent, render, RenderResult, screen } from '@testing-library/react'

import Input, { InputProps } from '../Input'

describe('Input', () => {
    const getProps = (): InputProps => ({
        value: '',
        onChange: jest.fn(),
    })

    const renderInput = (props = getProps()): RenderResult => {
        return render(<Input {...props} />)
    }

    it('Deve desenhar o componente corretamente', () => {
        renderInput()

        expect(document.body).toMatchSnapshot()
    })

    it('Deve chamar o onChange ao alterar o valor do input', () => {
        const props = {
            ...getProps(),
            placeholder: 'Your Text Here',
        }

        renderInput(props)
        const input = screen.getByPlaceholderText('Your Text Here')

        fireEvent.change(input, {
            target: {
                value: 'new input value',
            },
        })

        expect(props.onChange).toHaveBeenCalled()
    })
})
