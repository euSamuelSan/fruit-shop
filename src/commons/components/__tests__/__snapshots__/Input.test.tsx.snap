// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Deve desenhar o componente corretamente 1`] = `
<body>
  <div>
    <input
      class="sc-bczRLJ hKuxkL"
      type="text"
      value=""
    />
  </div>
</body>
`;
